pallets/ajuna-awesome-avatars/src/tests.rs

```diff
@@ -3987,3 +3987,227 @@ mod ipfs {
 			});
 	}
 }
+
+mod affiliates {
+	use super::*;
+	use pallet_ajuna_affiliates::traits::*;
+	use sp_runtime::bounded_vec;
+
+	#[test]
+	fn add_affiliate_to_account() {
+		let initial_balance = 1_000_000;
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.affiliators(&[ALICE])
+			.build()
+			.execute_with(|| {
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 0 }
+				);
+
+				assert_ok!(AAvatars::add_affiliation(RuntimeOrigin::signed(BOB), 0));
+
+				System::assert_last_event(mock::RuntimeEvent::Affiliates(
+					pallet_ajuna_affiliates::Event::AccountAffiliated { account: BOB, to: ALICE },
+				));
+
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliatees::<Test, AffiliatesInstance1>::get(BOB),
+					Some(bounded_vec![ALICE])
+				);
+
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::Affiliatable, affiliates: 1 }
+				);
+			});
+	}
+
+	#[test]
+	fn cannot_affiliate_to_non_enabled_account() {
+		let initial_balance = 1_000_000;
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.build()
+			.execute_with(|| {
+				assert_eq!(
+					pallet_ajuna_affiliates::Affiliators::<Test, AffiliatesInstance1>::get(ALICE),
+					AffiliatorState { status: AffiliatableStatus::NonAffiliatable, affiliates: 0 }
+				);
+
+				assert_noop!(
+					AAvatars::add_affiliation(RuntimeOrigin::signed(BOB), 0),
+					Error::<Test>::AffiliatorNotFound
+				);
+			});
+	}
+
+	#[ignore]
+	#[test]
+	fn enable_account_for_affiliation_free() {
+		let initial_balance = 1_000_000;
+		ExtBuilder::default()
+			.balances(&[(ALICE, initial_balance)])
+			.build()
+			.execute_with(|| {
+				assert_ok!(AAvatars::enable_affiliator(
+					RuntimeOrigin::signed(ALICE),
+					AffiliatorTarget::OneselfFree
```

Comment:
No, because this is something that is meant to be handled by the frontend, since you as an user may want to unlock that ability by paying.

Also the code does compile, it may be an issue with your IDE?